test/unit/executor/RV32I.cc

```diff
@@ -0,0 +1,100 @@
+#include "test_header.hh"
+#include "executor_test.hh"
+
+TEST(execute, add)
+{
+    simulationState.regs.set(2, 42);
+    sim::Instruction instr = {
+        1, // rs1
+        2, // rs2 
+        0,
+        1, // rd
+        0,
+        0,
+        sim::OpType::ADD,
+        41 // imm
+    };
+
+    ASSERT_EQ(simulationState.regs.get(1), 0);
+    executor.execute(instr, simulationState);
+    ASSERT_EQ(simulationState.regs.get(1), 42);
+}
+
+TEST(execute, sub)
+{
+    simulationState.regs.set(1, 15);
+    simulationState.regs.set(2, 5);
+    sim::Instruction instr = {
+        1, // rs1
+        2, // rs2 
+        0,
+        3, // rd
+        0,
+        0,
+        sim::OpType::SUB,
+        42 // imm
+    };
+
+    executor.execute(instr, simulationState);
+    ASSERT_EQ(simulationState.regs.get(3), 10);
+}
+
+TEST(execute, lw)
+{
+    // TODO
+}
+
+TEST(execute, sw)
+{
+    // TODO
+}
+
+TEST(execute, jal)
+{
+    simulationState.pc = 0x0;
+    sim::Instruction instr = {
+        1, // rs1
+        2, // rs2 
+        0,
+        3, // rd
+        0,
+        0,
+        sim::OpType::JAL,
+        0xFF // imm
+    };
+    executor.execute(instr, simulationState);
+    ASSERT_EQ(simulationState.npc, 0xFF);
+}
```

Comment:
I think you should also add a test for backward jump

Reply:
done